### PR TITLE
[PL-49125] Add Harness OIDC provider endpoint and OIDC audience settings

### PR DESCRIPTION
**Pull request by brian-f-harness**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: Add Harness OIDC provider endpoint and OIDC audience settings
* Jira/GitHub Issue numbers (if any): PL-49125
* Preview links/images (Internal contributors only): 
   * [AWS](https://662afbd11170693bf81f2e67--harness-developer.netlify.app/docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference)
   * [GCP](https://662afbd11170693bf81f2e67--harness-developer.netlify.app/docs/platform/connectors/cloud-providers/ref-cloud-providers/gcs-connector-settings-reference)

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [x] Successful preview build.
- [x] Code owner review.
- [x] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md [hunk 1]
```diff
@@ -423,13 +423,17 @@ Specify the credentials that enable Harness to connect your AWS account. There a
...
 * **Use IRSA:** Allows the Harness Kubernetes delegate in AWS EKS to use a specific IAM role when making authenticated requests to resources. By default, the Harness Kubernetes delegate uses a ClusterRoleBinding to the **default** service account; whereas, with this option, you can use AWS [IAM roles for service accounts (IRSA)](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) to associate a specific IAM role with the service account used by the Harness Kubernetes delegate. For instructions, go to [Use IRSA](/docs/platform/connectors/cloud-providers/add-aws-connector/#use-irsa).
-* **Use OIDC**: Connect to AWS with OIDC. In order to do this you will need to create an [OIDC identity provider](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html) in AWS and add it in a trust relationship with an IAM role you create that Harness will use to operate in AWS.
```
Comment (Raghav-Murali) [inline docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md:426]: Do we have a reference of the FF `CDS_AWS_OIDC_AUTHENTICATION` in this doc ? This flag has to be enabled for AWS OIDC feature.

File: docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md [hunk 2]
```diff
@@ -423,13 +423,17 @@ Specify the credentials that enable Harness to connect your AWS account. There a
...
-* **Use OIDC**: Connect to AWS with OIDC. In order to do this you will need to create an [OIDC identity provider](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html) in AWS and add it in a trust relationship with an IAM role you create that Harness will use to operate in AWS. 
+* **Use OIDC**: Connect to AWS with OIDC. To do this you will need to create an [OIDC identity provider](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html) in AWS and add it in a trust relationship with an IAM role you create that Harness will use to operate in AWS. Use the following Harness OIDC provider endpoint and OIDC audience settings to create your OIDC identity provider.
```
Comment (Raghav-Murali) [inline docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md:426]: Include the parent link of the doc - https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_oidc.html

File: docs/platform/connectors/cloud-providers/ref-cloud-providers/gcs-connector-settings-reference.md [hunk 3]
```diff
@@ -139,16 +139,21 @@ It is possible to create a connector with a non-existent delegate. This behavior
...
+
+Currently, the OIDC connectivity mode is not compatible with Google Cloud Functions. You can't deploy Google Cloud Functions with OIDC-enabled GCP connectors. Currently, this feature is behind the feature flag `PL_GCP_OIDC_AUTHENTICATION`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
+
+:::
+
 Select the **Connect through Harness Platform for OIDC** option to allow Harness to communicate directly with GCP through OIDC. This option uses OIDC authentication to access public cloud resources without secrets or credentials.
 
 Select the **Connect through Harness Delegate for OIDC** option to allow Harness Delegate to communicate directly with GCP through OIDC. This option uses OIDC authentication to access public cloud resources without secrets or credentials.
 
+To connect to GCP with OIDC, you must configure an [OIDC identity provider](https://cloud.google.com/identity-platform/docs/web/oidc#configuring_the_provider) in GCP and add it in a trust relationship with an IAM role you create that Harness will use to operate in GCP. Use the following Harness OIDC provider endpoint and OIDC audience settings to create your OIDC identity provider.
```
Comment (Raghav-Murali) [inline docs/platform/connectors/cloud-providers/ref-cloud-providers/gcs-connector-settings-reference.md:152]: Include this link instead - https://cloud.google.com/iam/docs/workload-identity-federation-with-other-providers.

There is no trust relationship in GCP. We need to connect the Service Account with relevant permissions to the Workload Identity Federation. So rewrite this line as -

To connect to GCP with OIDC, you must configure an [OIDC identity provider](https://cloud.google.com/identity-platform/docs/web/oidc#configuring_the_provider) in GCP and connect the service account with relevant permissions that Harness will use to operate in GCP.

File: docs/platform/connectors/cloud-providers/ref-cloud-providers/gcs-connector-settings-reference.md [hunk 4]
```diff
@@ -139,16 +139,21 @@ It is possible to create a connector with a non-existent delegate. This behavior
...
 
-:::note
+   * Harness OIDC provider endpoint: `https://app.harness.io/ng/api/oidc/account/<YOUR_ACCOUNT_ID>`
 
-Currently, the OIDC connectivity mode is not compatible with Google Cloud Functions. You can't deploy Google Cloud Functions with OIDC-enabled GCP connectors. Currently, this feature is behind the feature flag `PL_GCP_OIDC_AUTHENTICATION`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
-
-:::
+   * OIDC audience: `sts.amazonaws.com](https://iam.googleapis.com/projects/<YOUR_GCP_PROJECT_ID>/locations/global/workloadIdentityPools/<YOUR_WORKLOAD_POOL_ID>/providers/<YOUR_PROVIDER_ID>`
```
Comment (Raghav-Murali) [inline docs/platform/connectors/cloud-providers/ref-cloud-providers/gcs-connector-settings-reference.md:156]: No `sts.amazonaws.com` here.

**Review by Raghav-Murali — CHANGES_REQUESTED** [review]
Please address the comments.

File: docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md (hunk 1 above)
Comment (brian-f-harness) [reply docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md:426]: @Raghav-Murali Yes. I removed it with a commit. It was on line 434.

File: docs/platform/connectors/cloud-providers/ref-cloud-providers/gcs-connector-settings-reference.md (hunk 4 above)
Comment (brian-f-harness) [reply docs/platform/connectors/cloud-providers/ref-cloud-providers/gcs-connector-settings-reference.md:156]: copy-paste error. Thanks for catching it.

File: docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md (hunk 1 above)
Comment (brian-f-harness) [reply docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md:426]: @Raghav-Murali I also removed the other instances of the FF we had in docs.
cc @roshnisarangadharan @SushrutHarness @krishi0408

Comment (Raghav-Murali) [reply docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md:426]: @brian-f-harness This should not be removed. I was just asking whether a reference for this FF was provided earlier.

Comment (brian-f-harness) [reply docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md:426]: Thanks @Raghav-Murali I misread your comment.

**Review by Raghav-Murali — APPROVED** [review]

File: docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md [hunk 5]
```diff
@@ -423,14 +423,17 @@ Specify the credentials that enable Harness to connect your AWS account. There a
...
-* **Use OIDC**: Connect to AWS with OIDC. In order to do this you will need to create an [OIDC identity provider](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc.html) in AWS and add it in a trust relationship with an IAM role you create that Harness will use to operate in AWS. 
+* **Use OIDC**: Connect to AWS with OIDC. To do this you will need to create an [OIDC identity provider](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_oidc.html) in AWS and add it in a trust relationship with an IAM role you create that Harness will use to operate in AWS. Use the following Harness OIDC provider endpoint and OIDC audience settings to create your OIDC identity provider.
```
Comment (douglas-j-bothwell) [inline docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md:426]:
```suggestion
* **Use OIDC**: Connect to AWS with OIDC. To do this, you need to create an [OIDC identity provider](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_oidc.html) in AWS. Then you need to add it in a trust relationship with an IAM role you create that Harness will use to operate in AWS. Use the following Harness OIDC provider endpoint and OIDC audience settings to create your OIDC identity provider.
```

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 6]
```diff
@@ -6172,8 +6172,11 @@ Yes, one can deploy using custom artifact without using the S3 artifact. Please
 
 #### How does the AWS connector currently handle OIDC authentication?
 
-Users can now leverage the AWS connector to configure OIDC authentication. This only leverages the account ID parameter when we send the token payload to the AWS STS service. This feature is provided behind the feature flag `FF:CDS_AWS_OIDC_AUTHENTICATION`
-Please read more on this in the following [Documentation](https://developer.harness.io/docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference/#harness-aws-connector-settings)
+Users can now leverage the AWS connector to configure OIDC authentication.
+
+This feature is currently behind the feature flag `CDS_AWS_OIDC_AUTHENTICATION`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
+
+This only leverages the account ID parameter when we send the token payload to the AWS STS service. For more information, go to [Harness AWS connector settings](https://developer.harness.io/docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference/#harness-aws-connector-settings).
```
Comment (douglas-j-bothwell) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6179]:
```suggestion
This only leverages the account ID parameter when Harness sends the token payload to the AWS STS service. For more information, go to [Harness AWS connector settings](https://developer.harness.io/docs/platform/connectors/cloud-providers/ref-cloud-providers/aws-connector-settings-reference/#harness-aws-connector-settings).
```

**Review by douglas-j-bothwell — DISMISSED** [review]
A few sugs otherwise LGTM

**Pull request merged**